### PR TITLE
Update lint-history to add callback argument

### PR DESCRIPTION
**Pull request by LunarLanding**
Together with the relevant callback, this allows both filtering on filenames and transforming blobs without having to spawn a different process, by importing the corresponding tool's module. This is essential for performance with python-based tools.
The new code is strictly backwards compatible.

### CODE REVIEW

File: contrib/filter-repo-demos/lint-history
Comment (huyz) [inline contrib/filter-repo-demos/lint-history:113]: More consistent styling:
```suggestion
if bool(lint_args.command) + bool(lint_args.callback) != 1:
```

**Review by rkwjdgus9880 — APPROVED** [review]

File: contrib/filter-repo-demos/lint-history
Comment (newren) [inline contrib/filter-repo-demos/lint-history:114]: Why the capital 'O'?

File: contrib/filter-repo-demos/lint-history
Comment (newren) [inline contrib/filter-repo-demos/lint-history:152]: Leftover debugging modification?

File: contrib/filter-repo-demos/lint-history
Comment (newren) [inline contrib/filter-repo-demos/lint-history:136]: I don't understand; what good can this do since blob will still refer to the original value?  I would have thought I'd see a `blob = callback...`; am I missing something?

File: contrib/filter-repo-demos/lint-history
Comment (newren) [inline contrib/filter-repo-demos/lint-history:176]: Why a space after the `=` but not before?  I've seen some folks that like no spaces, and some that like spaces on each side, but never this.  Regardless, style in the rest of the file is a space on each side.

File: contrib/filter-repo-demos/lint-history
Comment (newren) [reply contrib/filter-repo-demos/lint-history:113]: Agree with @huyz here.

File: contrib/filter-repo-demos/lint-history
Comment (newren) [inline contrib/filter-repo-demos/lint-history:83]: "programwill" has a missing space.  But perhaps replace the new phrase with "when not using --callback"?

**Review by newren — CHANGES_REQUESTED** [review]
In addition to various comments on the patch:

Is "LunarLanding" your real name?  I suspect it is not, but I need your real name in author/committer/signoff.

Similarly, I need a real email in author/committer/signoff; @users.noreply.github.com addresses do not qualify.

Please wrap lines in the commit message at 72 characters.

Please add your signoff (https://github.com/git/git/blob/master/Documentation/SubmittingPatches#L351) to your commit.

**Comment by LunarLanding** [comment]
Please take over the pull request (ie drop history of my contribution, just copy the code and modify as fit). I cannot at this time continue this effort.

**Comment by newren** [comment]
> Please take over the pull request (ie drop history of my contribution, just copy the code and modify as fit). I cannot at this time continue this effort.

Understood; sorry for the delay in getting back to you.  Without proper sign-off, I don't think it'd be appropriate to "fix up" the changes.  I can't accept them without that, so we'll just drop this pull request for now and if you want to come back to it (or someone independently wants the same thing and generates it from scratch), then we'll proceed.

Thanks for your interest and taking the time to send the contribution in.  I may mention this in an extended list of examples I've been contemplating for people to take a look at.

**Comment by newren** [comment]
The new --file-info-callback in git-filter-repo makes this kind of functionality available to users.

**Pull request closed without merging**
